src/network/address.rs

```diff
@@ -67,25 +67,17 @@ impl Address {
     }
 }
 
-fn addr_to_be(addr: [u16; 8]) -> [u16; 8] {
-    // consensus_encode always encodes in LE, and we want to encode in BE.
-    // this utility fn swap bytes before encoding so that the encoded result will be BE
-    let mut result = addr;
-    for word in &mut result {
-        *word = word.swap_bytes();
-    }
-    result
-}
-
 impl Encodable for Address {
     #[inline]
     fn consensus_encode<S: io::Write>(&self, mut s: S) -> Result<usize, io::Error> {
-        let len = self.services.consensus_encode(&mut s)?
-            + addr_to_be(self.address).consensus_encode(&mut s)?
+        let mut len = self.services.consensus_encode(&mut s)?;
+
+        for word in &self.address {
+            len += io::Write::write(&mut s, &word.to_be_bytes())?;
```

Comment:
Just realized we have the same issue as with `read_exact`, we need `write_all()` here. Also while changing you could use `s.write_all()` instead, no need to repeat the whole trait since it can't get confused anyway thanks to generics.

Reply:
Legend man, late is better than never :) Will fix as suggested.